### ISSUE TITLE
documentation needs to be updated to reflect new I_ext syntax

### ISSUE THREAD

**Issue by sg-s** [sg-s/xolotl#141]

**Comment by alec-hoyland** [comment]
Not sure that I_ext implementation works at the moment
```matlab
% works fine
x     = xolotl;
x.add('HH', 'compartment', 'Cm', 10, 'A', 0.01);
x.HH.add('liu/NaV', 'gbar', 1000, 'E', 50);
x.HH.add('liu/Kd', 'gbar', 300, 'E', -80);
x.HH.add('Leak', 'gbar', 1, 'E', -50);
x.t_end = 5e3; x.dt = 0.1;
x.I_ext = 0.1;
V = x.integrate;
% works fine
x.I_ext = 0.1*(1:length(round(x.t_end/x.dt));
% tries to create a massive array 
x.replicate('HH', 2)
x.I_ext = 0.1 * ones(100000,2);
V = x.integrate % error here
x.I_ext = [0.1 0.1]
V = x.integrate % produces the same error

Error using mexBridge68599f
Requested 1x18446744071562067968 (17179869184.0GB) array exceeds maximum array size preference. Creation of arrays greater than this
limit may take a long time and cause MATLAB to become unresponsive. See array size limit or preference panel for more information.

Error in xolotl/integrate (line 78)
[results{1:nargout+1}] = f(arguments,self.I_ext,self.V_clamp);
```

**Comment by sg-s** [comment]
also `length(round(x.t_end/x.dt))` is pretty nonsensical, because it's just `1`

**Comment by alec-hoyland** [comment]
Yeah, I'm a little frazzled. You're totally right. 

What I mean to say is:

Scalar `I_ext` works fine.
Vector `I_ext` works fine.
Matrix `I_ext` overflows memory.

**Comment by sg-s** [comment]
are you sure?

```
x.replicate('HH', 2)
x.I_ext = 0.1 * ones(100000,2);
```

will not work, because you have 3 compartments.

**Comment by alec-hoyland** [comment]
No. `replicate` makes `n` compartments total.

```matlab
>> x     = xolotl;
x.add('HH', 'compartment', 'Cm', 10, 'A', 0.01);
x.HH.add('liu/NaV', 'gbar', 1000, 'E', 50);
x.HH.add('liu/Kd', 'gbar', 300, 'E', -80);
x.HH.add('Leak', 'gbar', 1, 'E', -50);
>> x.replicate('HH',2)
>> xolotl object with 
---------------------
+ HH  
  > Kd (g=300, E=-80)
  > Leak (g=1, E=-50)
  > NaV (g=1000, E=50)
---------------------
+ HH2  
  > Kd (g=300, E=-80)
  > Leak (g=1, E=-50)
  > NaV (g=1000, E=50)
---------------------
```

The issue I think is line 173 in `xolotl.m` where you use `repmat`. I think you mean `reshape`.

**Comment by sg-s** [comment]
OK, that's a bug with matrices. will fix.

**Comment by sg-s** [comment]
OK, should be all right now. the same issue also affected `V_clamp`

**Comment by alec-hoyland** [comment]
Closed with #147
